### PR TITLE
Alert

### LINKED ISSUE #40: Alert without saving
# Feature Request

## Summary

When the user tried to go to the home page without saving their notes, or close the task list without saving it, the website did not send any message to alert the user,.

## Details

When the user tries to exit their notes without saving, or tries to reload the page, a message will notify the user if they forgot to save their notes. And also when the user close the task list without saving it, the user will receive a message.

### PR DESCRIPTION
**Pull request by hdsouza13**
*Description*

The user forgot to save their data in the Dev Journal, or in the Task List in the calendar, so the user will receive an alert message if they try to go to the home page or try to reload the page without saving it. If the user write something and then erase everything and try to go to the home page, no alert message will show.

*Changes Made*

- Implemented the function in the dev-jounal.js and task-list.js

- Implemented in the terminal.js a event, so the alert message will not show when the user save using the terminal

*Testing*

- I tested using the click buttom of save and the commands in the terminal.

- cd ..

*Screenshots*

Alert1
Alert2
Alert3
Alert4
Alert5
Alert6

*Pull Request Review*
- [x]  All automated tests pass locally.
- [x]  Manual testing has been performed according to the provided instructions.
- [x]  Code follows the project's coding conventions and standards.
- [x]  Documentation has been updated to reflect any relevant changes.
- []  This PR has been reviewed by at least 2 of the team members.
- [x]  It passes all the appropriate Linters

### CODE REVIEW

**Review by vishaal-gaddipati — APPROVED** [review]
Just tested on my end briefly and the alert seems to function as expected. Looks good to merge into dev once the merge conflicts are handled.

**Review by Arnav710 — APPROVED** [review]
Resolved the merge conflicts. When there are unsaved changes in the calendar or dev journal, the `alert` shows up as intended.

**Pull request merged**
